beacon-chain/sync/backfill/metrics.go

```diff
@@ -21,86 +21,117 @@ var (
 			Help: "Number of batches that are ready to be imported once they can be connected to the existing chain.",
 		},
 	)
-	backfillRemainingBatches = promauto.NewGauge(
+	batchesRemaining = promauto.NewGauge(
 		prometheus.GaugeOpts{
 			Name: "backfill_remaining_batches",
 			Help: "Backfill remaining batches.",
 		},
 	)
-	backfillBatchesImported = promauto.NewCounter(
+	batchesImported = promauto.NewCounter(
 		prometheus.CounterOpts{
 			Name: "backfill_batches_imported",
 			Help: "Number of backfill batches downloaded and imported.",
 		},
 	)
-	backfillBlocksApproximateBytes = promauto.NewCounter(
+
+	backfillBatchTimeWaiting = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "backfill_batch_time_waiting",
+			Help:    "Time batch waited for a suitable peer.",
+			Buckets: []float64{50, 100, 300, 1000, 2000},
+		},
+	)
+	backfillBatchTimeRoundtrip = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "backfill_batch_time_roundtrip",
+			Help:    "Total time to import batch, from first scheduled to imported.",
+			Buckets: []float64{400, 800, 1600, 3200, 6400, 12800},
+		},
+	)
+
+	blockDownloadCount = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "backfill_blocks_download_count",
+			Help: "Number of BeaconBlock values downloaded from peers for backfill.",
+		},
+	)
+	blockDownloadBytesApprox = promauto.NewCounter(
 		prometheus.CounterOpts{
 			Name: "backfill_blocks_bytes_downloaded",
 			Help: "BeaconBlock bytes downloaded from peers for backfill.",
 		},
 	)
-	backfillBlobsApproximateBytes = promauto.NewCounter(
-		prometheus.CounterOpts{
-			Name: "backfill_blobs_bytes_downloaded",
-			Help: "BlobSidecar bytes downloaded from peers for backfill.",
+	blockDownloadMs = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "backfill_batch_blocks_time_download",
+			Help:    "BeaconBlock download time, in ms.",
+			Buckets: []float64{100, 300, 1000, 2000, 4000, 8000},
+		},
+	)
+	blockVerifyMs = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "backfill_batch_time_verify",
+			Help:    "BeaconBlock verification time, in ms.",
+			Buckets: []float64{100, 300, 1000, 2000, 4000, 8000},
```

Comment:
Bucket distribution not accurate:

<img width="1896" height="351" alt="Image" src="https://github.com/user-attachments/assets/e88496dc-9354-4287-b45c-d40eb8637932" />

Reply:
By not accurate do you mean that everything is falling into one bucket? The values eg .111 makes me think we might be reporting decimal of seconds as ms (so .111 ms is actually 111ms).

Reply:
Does not seem to be the issue though:
```
blockVerifyMs.Observe(float64(time.Since(dlt).Milliseconds()))
```

Reply:
Could you post the source of this grafana panel?

Reply:
It's not a grafana panel actually, it's only the "drilldown" of this metric.

<img width="503" height="401" alt="image" src="https://github.com/user-attachments/assets/09f546c4-00a9-4b43-ba23-35058765cdd9" />

>By not accurate do you mean that everything is falling into one bucket?

Yes.

Reply:
I am seeing the verification take 6ms on average. These buckets are way off! It would never take 8000ms to verify the signature. I'd recommend something like

```suggestion
			Buckets: []float64{3, 5, 10, 20, 100, 200},
```